cmd/agent/subcommands/streamlogs/command.go

```diff
@@ -97,3 +136,21 @@ func streamRequest(url string, body []byte, onChunk func([]byte)) error {
 	}
 	return e
 }
+
+func writeToFile(filePath, message string) (*os.File, *bufio.Writer, error) {
+	// Open the file for writing, create it if it does not exist.
+	f, err := os.OpenFile(filePath, os.O_APPEND|os.O_CREATE|os.O_WRONLY, 0644)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	// Use a buffered writer to minimize direct writes to disk.
+	bufWriter := bufio.NewWriter(f)
+
+	_, err = bufWriter.WriteString(message + "\n")
```

Comment:
Another thing I just noticed - the chunk returned from the stream is not guaranteed to be a single log line. This is why we use `fmt.Print` and not `fmt.Println` above. It's just an arbitrary chunk of bytes. So you don't need to newline delimit this. 

In fact - the messages are already [newline delimited](https://github.com/DataDog/datadog-agent/blob/b386591b15faab716428d43633f83324ca0ff610/pkg/logs/diagnostic/format.go#L42)